src/aspire/volume/__init__.py

```diff
@@ -1,16 +1,164 @@
 import numpy as np
 
+from aspire.image import Image
 from aspire.nufft import Plan
 from aspire.utils import ensure
 from aspire.utils.coor_trans import grid_2d
 from aspire.utils.fft import centered_fft2, centered_ifft2
 from aspire.utils.matlab_compat import m_flatten, m_reshape
+from aspire.utils.preprocess import downsample
 
 
 class Volume:
     """
-    stack of volumes
+    Volume is an N x L x L x L array, along with associated utility methods.
     """
+    def __init__(self, data):
+        """
+        Create a volume initialized with data.
+
+        Volumes should be N x L x L x L,
+        or L x L x L which implies N=1.
+
+        :param data: Volume data
+
+        :return: A volume instance.
+        """
+
+        if data.ndim == 3:
+            data = data[np.newaxis, :, :, :]
+
+        ensure(data.ndim == 4,
+               'Volume data should be ndarray with shape NxLxLxL'
+               ' or LxLxL.')
+
+        ensure(data.shape[1] == data.shape[2] == data.shape[3],
+               'Only cubed ndarrays are supported.')
+
+        self.data = data
+        self.N = self.data.shape[0]
```

Comment:
I was saving this, the conversion of `VolumeInstance.data` usage to `asnumpy()`, and a few other things for a pass just cleaning up Volume class, but I changed `n_vols` now since it was only a few lines to change. The Volume class is still really green, so it is a good time for you to make aesthetic decisions.

Reply:
Ok. I will take a closer look.